src/main/java/build/buildfarm/server/ByteStreamService.java

```diff
@@ -259,7 +263,8 @@ private void writeBlob(
           active_write_requests.remove(writeResourceName);
           Digest blobDigest = Digests.computeDigest(data);
           if (!blobDigest.equals(digest)) {
-            responseObserver.onError(new StatusException(Status.INVALID_ARGUMENT));
+            String description = String.format("Digest mismatch %s <-> %s", blobDigest.getHash(), digest.getHash());
```

Comment:
Prefer `Digests.toString(blobDigest)` etc here - the equals comparison occurs between both hash and size, and we have the toString mechanism representing all other presentations of digests